dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -89,6 +98,41 @@ async def snapshot(self, req):
             success=True, message="hello", snapshot=snapshot
         )
 
+    @routes.get("/api/component_activities")
+    async def get_component_activities(self, req) -> aiohttp.web.Response:
+        # Get activity information for driver
+        driver_activity_info = await self._get_job_activity_info()
+
+        resp = {"driver": dataclasses.asdict(driver_activity_info)}
```

Comment:
Why don't we not include `driver`? This will make the schema management confusing.

Let's just resp = asdict(driver_activity_info)?

Reply:
We wanted the response model of this endpoint to be `Dict[str, RayActivityResponse]` rather than including the component name in `RayActivityResponse` so it is easier to loop through and maintain uniqueness for the various component types. The schema for this endpoint response has a wildcard for the component types (instead of enumerating options) to allow external component types once we add a hook (coming in next PR).

Reply:
In this case, we need to change dashboard/modules/snapshot/component_activities_schema.json? The filename indicates it's the schema for the endpoint `component_activities `, but it is actually different from real schema.

Reply:
Or even better way can be just 

```python3
@dataclasses.dataclass
class RayActivityResponse:
    """
    Dataclass used to inform if a particular Ray component can be considered
    active, and metadata about observation.
    """

    # Whether the corresponding Ray component is considered active
    is_active: bool
    ...

@dataclasses.dataclass
class RayActivityResponse:
    activity_per_component: Dict[str, DriverActivityInfo]
```
?